adi/ad7944.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2020-2024 Analog Devices, Inc.
+#
+# SPDX short identifier: ADIBSD
+
+import numpy as np
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class ad7944(rx, context_manager):
+    """AD7944, 14-bit, successive approximation analog-to-digital
+    converter (SAR ADC) with sample rates up to 2.5 MSPS"""
+
+    _compatible_parts = ["ad7944"]
+    _device_name = ""
+    _rx_data_type = np.uint16
```

Comment:
FYI, the HDL team has decided to try to standardize on using 32-bit words for SPI offload support.

So this means that the data type will be 16-bit if not using SPI offload but 32-bit if using SPI offload.